src/pathspec_user_guide.md

```diff
@@ -0,0 +1,444 @@
+
+### PathSpec in Validation
+PathSpec's string form is used for [Request Validation](https://linkedin.github.io/rest.li/Validation-in-Rest_li) in Rest.Li Resource. A string path can be added in annotation such as "CreateOnly" and "ReadOnly". For example:
+```java
+@CreateOnly({"/id", "/EXIF"})
+public class PhotoResource extends CollectionResourceTemplate<Long, Photo>
+{
+    ...
+}
+```
+
+This string is then parsed by [RestLiAnnotationReader](https://github.com/linkedin/rest.li/blob/master/restli-server/src/main/java/com/linkedin/restli/internal/server/model/RestLiAnnotationReader.java), and [DataSchemaUtil](https://github.com/linkedin/rest.li/blob/master/data/src/main/java/com/linkedin/data/schema/DataSchemaUtil.java) will do corresponding validation against this path within the DataMap.
+
+You can also invoke [RestliDataValidator](https://github.com/linkedin/rest.li/blob/master/restli-common/src/main/java/com/linkedin/restli/common/validation/RestLiDataValidator.java) and pass PathSpec string to it. [TestRestLiValidation.java](https://github.com/linkedin/rest.li/blob/master/restli-int-test/src/test/java/com/linkedin/restli/examples/TestRestLiValidation.java)'s testCustomValidatorMap() is such an example
+```java
+    ...
+
+    Map<String, List<String>> annotations = new HashMap<>();
+    annotations.put("createOnly", Arrays.asList("stringB", 
+        "intB", "UnionFieldWithInlineRecord/com.linkedin.restli.examples.greetings.api.myRecord/foo2", "MapWithTyperefs/*/id"));
+    annotations.put("readOnly", Arrays.asList("stringA", 
+        "intA", "UnionFieldWithInlineRecord/com.linkedin.restli.examples.greetings.api.myRecord/foo1", "ArrayWithInlineRecord/*/bar1", "validationDemoNext/stringB", "validationDemoNext/UnionFieldWithInlineRecord"));
+    ...
+
+    validator = new RestLiDataValidator(annotations, ValidationDemo.class, ResourceMethod.CREATE, validatorClassMap);
+
+```
+
+## PathSec Semantics
+
+PathSpec has been defined as a path to a component within a complex data object path. This provides a way to traverse the Pegasus data object. The abstract data object, in most case, is a form of [DataMap](https://github.com/linkedin/rest.li/blob/master/data/src/main/java/com/linkedin/data/DataMap.java) internally in RestLi framework, but PathSpec should be meaningful for the same data object in other forms, for example, a json representing the same data object.
+
+The PathSpec format is represented by separators(`'/'`) and segments in between. 
+```
+/demoRecord/innerRecordField/nestedInnerRecordField
+```
+
+The path segment for some types, such as array, could have some attributes, which following the `&` sign, 
```

Comment:
you are right and I am aware of that. But I don't think it meaningful to introduce non-reserved attribute since they are not interpreted anyway. It will add to confusion.

But I have re-worded this part.